miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebChromeClient.kt

```diff
@@ -85,4 +90,64 @@ internal class MiniAppWebChromeClient(
     internal fun doInjection(webView: WebView) {
         webView.evaluateJavascript(bridgeJs) {}
     }
+
+    //region fullscreen video
+    @VisibleForTesting
+    internal var customView: View? = null
+    private var customViewCallback: CustomViewCallback? = null
+    private var originalOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+    private var originalSystemUiVisibility = View.SYSTEM_UI_FLAG_VISIBLE
+    private val fullScreenFlag = View.SYSTEM_UI_FLAG_FULLSCREEN or
+            View.SYSTEM_UI_FLAG_HIDE_NAVIGATION or
+            View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION or
+            View.SYSTEM_UI_FLAG_IMMERSIVE
+
+    override fun onShowCustomView(paramView: View?, paramCustomViewCallback: CustomViewCallback?) {
+        if (customView != null) {
+            onHideCustomView()
+            return
+        }
+        customView = paramView
+        if (context is Activity) {
+            context.apply {
+                originalSystemUiVisibility = window.decorView.systemUiVisibility
+                originalOrientation = requestedOrientation
+                customViewCallback = paramCustomViewCallback
+                (window.decorView as FrameLayout).addView(customView, FrameLayout.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT))
+                window.decorView.systemUiVisibility = fullScreenFlag
+                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
+                customView?.setOnSystemUiVisibilityChangeListener { updateControls() }
+            }
+        }
+    }
+
+    override fun onHideCustomView() {
+        if (context is Activity) {
+            context.apply {
+                (window.decorView as FrameLayout).removeView(customView)
+                customView = null
+                window.decorView.systemUiVisibility = originalSystemUiVisibility
+                requestedOrientation = originalOrientation
+                customViewCallback!!.onCustomViewHidden()
+                customViewCallback = null
+                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER
+            }
+        }
+    }
+
+    private fun updateControls() {
+        val params = (customView?.layoutParams as FrameLayout.LayoutParams).apply {
+            bottomMargin = 0
+            topMargin = 0
+            leftMargin = 0
+            rightMargin = 0
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = ViewGroup.LayoutParams.MATCH_PARENT
+        }
+        customView?.layoutParams = params
+        if (context is Activity)
+            context.window.decorView.systemUiVisibility = fullScreenFlag
```

Comment:
It seems like this is causing the system controls to disappear almost immediately (after swiping down in full screen), so there's no time to actually use the controls. Can we add a delay so that the controls are hidden after 1 or 2 seconds?

Reply:
It's my intention to hide the android system navigator due to the overlap UI between system bottom bar & media bar.
Users can only interact with system navigator when they swipe status bar or exit fullscreen.

Reply:
@minh-rakuten Yes, that's what I mean. The system controls should be hidden normally, but currently even when swiping down from the status bar, the controls and status bar appear for only a split second and then disappear almost immediately. We need to let them stay visible long enough for the user to interact with them.

Reply:
You're right. That issue happens on Android. I have solved it by update system ui flag.